pandas/tests/frame/methods/test_replace.py

```diff
@@ -661,6 +661,18 @@ def test_replace_simple_nested_dict_with_nonexistent_value(self):
         result = df.replace({"col": {-1: "-", 1: "a", 4: "b"}})
         tm.assert_frame_equal(expected, result)
 
+    def test_replace_numpy_nan(self):
+        # GH#45725 ensure np.nan can be replaced with pd.NA
+        df = pd.DataFrame({"A": [np.nan], "B": [np.nan]}, dtype=object)
+        result = df.replace({np.nan: pd.NA})
+        expected = pd.DataFrame({"A": [pd.NA], "B": [pd.NA]}, dtype=object)
+        tm.assert_frame_equal(result, expected)
+
+        # same thing but with None
```

Comment:
Could you use `pytest.mark.parameterize` with `pd.NA/None` (same for the other test)?

Reply:
Done.